docs/src/formula.md

```diff
@@ -14,167 +14,191 @@ fields with possibly heterogeneous types.  One of the primary goals of
 `StatsModels` is to make it simpler to transform tabular data into matrix format
 suitable for statistical modeling.
 
-At the moment, "tabular data" means an `AbstractDataFrame`.  Ultimately, the
-goal is to support any tabular data format that adheres to a minimal API,
-**regardless of backend**.
-
-## The `Formula` type
-
-The basic conceptual tool for this is the `Formula`, which has a left side and a
-right side, separated by `~`. Formulas are constructed using the `@formula` macro:
-
-```jldoctest
-julia> @formula(y ~ 1 + a)
-Formula: y ~ 1 + a
+At the moment, "tabular data" means a
+[Tables.jl](https://github.com/JuliaData/Tables.jl) table, which will be
+materialized as a `Tables.ColumnTable` (a `NamedTuple` of column vectors).  Work
+on first-class support for streaming/row-oriented tables is ongoing.
+
+## The `@formula` DSL
+
+StatsModels implements the [`@formula`](@ref) domain-specific language (DSL) for
+describing table-to-matrix transformations.  This DSL is designed to be familiar
+to users of other statistical software, while also taking advantage of Julia's
+unique strengths to be fast and flexible.
+
+A basic formula is composed of individual *terms*—symbols which refer to data
+columns, or literal numbers `0` or `1`—combined by `+`, `&`, `*`, and (at the
+top level) `~`.
+
+!!! note 
+
+    The `@formula` macro **must** be called with parentheses to ensure that
+    the formula is parsed properly.
+
+Here is an example of the `@formula` in action:
+
+```@repl 1
```

Comment:
It would be nice to turn these into `jldoctest` blocks so that any change in output it detected. Can be done later though.

Reply:
shouldn't be too hard.  I think there's a documenter setting that will actually generate the doctest format, or I can just copy and paste from the markdown output

Reply:
okay, all doctests now!  the added bonus of that is that the actual output is included in the source files so it's a net win for sure